modules/rte_mis_mail/rte_mis_mail.module

```diff
@@ -0,0 +1,411 @@
+<?php
+
+/**
+ * @file
+ * Primary module hooks for registration_otp module.
+ */
+
+use Drupal\Core\Ajax\AjaxResponse;
+use Drupal\Core\Ajax\HtmlCommand;
+use Drupal\Core\Ajax\ReplaceCommand;
+use Drupal\Core\Cache\Cache;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\session_limit\Form\SessionLimitForm;
+
+/**
+ * Implements hook_form_id_alter().
+ */
+function rte_mis_mail_form_user_register_form_alter(array &$form, FormStateInterface &$form_state) {
+  // Load the configuration object.
+  $config = \Drupal::config('email_sms_form.settings');
+  // Get the status of verification.
+  $verify = $config->get('email_verify');
+  $form['#cache']['tags'] = Cache::mergeTags($form['#cache']['tags'], $config->getCacheTags());
+
+  if ($verify == TRUE) {
+    $form['email_container'] = [
+      '#type' => 'container',
+      '#attributes' => ['id' => ['email-container']],
+    ];
+    // Hidden field for controlling the display.
+    $form['display_handler'] = [
+      '#type' => 'hidden',
+      '#default_value' => FALSE,
+    ];
+    // Move the existing email field into the container.
+    $form['email_container']['account']['mail'] = $form['account']['mail'];
+    // Remove the original placement of the email field.
+    unset($form['account']['mail']);
+    // Add the Verify Email button inside the container.
+    $form['email_container']['verify_email'] = [
+      '#type' => 'button',
+      '#value' => t('Verify Email'),
+      '#ajax' => [
+        'callback' => 'rte_mis_mail_email_verify',
+        'wrapper' => 'email-container',
+        'effect' => 'fade',
+      ],
+    ];
+    // Container that shall contain the otp related fields and buttons.
+    $form['email_container']['otp_container'] = [
+      '#type' => 'container',
+      '#attributes' => ['id' => 'otp-container'],
+      '#states' => [
+        'visible' => [
+          ':input[name="display_handler"]' => ['value' => TRUE],
+        ],
+      ],
+    ];
+    // Otp field.
+    $form['email_container']['otp_container']['field_otp'] = [
+      '#type' => 'textfield',
+      '#title' => t('Enter OTP'),
+      '#attributes' => [
+        'autocomplete' => 'off',
+      ],
+    ];
+    // Otp submit button.
+    $form['email_container']['otp_container']['otp_submit_button'] = [
+      '#type' => 'button',
+      '#value' => t('Submit OTP'),
+      '#ajax' => [
+        'callback' => 'rte_mis_mail_otp_submit',
+        'wrapper' => 'account-creation-button email-container',
+        'effect' => 'fade',
+      ],
+    ];
+    // Otp resend.
+    $form['email_container']['otp_container']['otp_resend_button'] = [
+      '#type' => 'button',
+      '#value' => t('Resend OTP'),
+      '#ajax' => [
+        'callback' => 'rte_mis_mail_otp_resend',
+        'wrapper' => 'email-container',
+        'effect' => 'fade',
+      ],
+    ];
```

Comment:
Can you please validate on scenario where some required fields are empty and you click resend OTP. Will it throw all the errors or just a message saying OTP sent?

Reply:
Sure. I have validated the above scenario and its working fine.